tcms/testruns/static/testruns/js/get.js

```diff
@@ -180,13 +180,36 @@ function renderTestExecutions(testExecutions) {
     })
 
     treeViewBind();
+    $('.test-executions-count').html(testExecutions.length)
     renderAdditionalInformation(testExecutions, testCaseIds)
 }
 
 function renderAdditionalInformation(testExecutions, testExecutionCaseIds) {
-    $('.test-executions-count').html(testExecutions.length);
 
     renderTestCaseInformation(testExecutions, testExecutionCaseIds)
+
+    testExecutions.forEach(testExecution => {
+        jsonRPC('TestExecution.history', testExecution.id, history => {
+            const liHistory = $(`.test-execution-${testExecution.id} .history-container`)
+            history.forEach(h => {
+                liHistory.append(renderHistoryEntry(h))
+            })
+        })
+    })
+}
+
+function renderHistoryEntry(historyEntry) {
+    if (!historyEntry.history_change_reason) {
+        return ''
+    }
+
+    const template = $($('#history-entry')[0].content.cloneNode(true))
+
+    template.find('.history-date').html(historyEntry.history_date)
+    template.find('.history-user').html('TODO: user')
```

Comment:
what is this for  ?

Reply:
ah, yes. the history API returns entries in format like this:
```json
{
   "assignee_id":1,
   "build_id":1,
   "case_id":1,
   "case_text_version":1174,
   "​​close_date":"2019-02-11T19:06:53.509",
   "​​history_change_reason":"",
   "​​history_date":"2020-09-26T13:13:00.565",
   "​​history_id":6187,
   "​​history_type":"~",
   "​​history_user_id":15,
   "​​id":1,
   "​​run_id":1,
   "​​sortkey":20,
   "​​status_id":7,
   "​​tested_by_id":15
}
```
we have `user_id`, but no username. The old UI shows the username of the user that made the changes, but I guess this would mean making another API call. don't know if there isn't an easier option, so left this as a reminder to come back to this

Reply:
.select_related('history_user__username') on the query set.

Reply:
```
execution.history.select_related('history_user__username').all()[0]
> django.core.exceptions.FieldError: Non-relational field given in select_related: 'username'. Choices are: (none)
```

Reply:
it worked when I passed `history_user__username` directly to values. However, this meant that I also had to enumerate the other fields I am using, so that they can be returned